content/assignments/techjobs-console/_index.md

```diff
@@ -0,0 +1,477 @@
+   of your job.
+
+Let's look at each of these.
+
+### The `RunProgram()` Method
+
+The logic within `RunProgram()` presents menus in turn, and based on the
+user's choice, takes appropriate action.
+
+It begins by declaring two local variables: `actionChoices` and
+`columnChoices`. These contain information relating to the menus that
+we'll display, and we'll look at them in more detail later.
+
+Next, we notice a `while loop` that starts `while (true)`. While we usually
+want to avoid creating infinite loops, we have a good reason for doing so in
+this case! We want our application to continually run until the user decides
+they want to quit. The simplest way to do this is to loop forever. When the
+user wants to quit our app, they can enter `x` at the initial `View jobs by` prompt. 
+
+{{% notice blue "Note" "rocket" %}}
+
+   There are two ways to stop a running app. Either option works so you can pick the one that works best for you!
+   
+   1. Use the IDE.  Click Visual Studio's *Stop* icon.  The light red square that replaces the green *Run* triangle once an app is running.
+   1. Use the terminal. Press *ctrl+C* (a widely-known command to kill a console application). This will work in any terminal context, and not just for our console program in Visual Studio
+
+{{% /notice %}}
+
+The `RunProgram()` method can be summarized as follows:
+
+1. Present the user with choices on how to view data: *list* or *search*.
+1. Based on that choice, prompt them for the column to apply the choice to. In
+   the case of a search, we also ask for a search term.
+1. Carry out the request to the `JobData` class via one of its public
+   methods.
+1. Display the results of the request.
+1. Repeat.
+
+`RunProgram()` simulates a *query* to an external source:
+
+1. We ask the method for data that originates from a non-C# source.
+1. The method parses and filters that data.
+1. The method presents the data in a useful manner.
+
+### The `GetUserSelection()` Method
+
+The `GetUserSelection()` method takes in a string to display above the
+menu, to provide context for what they are being asked. It also takes in
+a `Dictionary` with string keys and string values. How is this used? What
+will this `Dictionary` contain when the method runs?
+
+To figure this out, right-click on the method name and select *Find (All)
+References*. This will open a pane and display each location in the program
+where `GetUserSelection()` is called. The first such usage is the first
+line of the main `while loop`:
+
+```csharp
+   string actionChoice = GetUserSelection("View Jobs", actionChoices);
+```
+
+What is this `Dictionary` named `actionChoices`? If we look a few lines
+above, we see:
+
+```csharp {linenos=true,linenostart=13}
+   // Top-level menu options
+   Dictionary<string, string> actionChoices = new Dictionary<string, string>();
+   actionChoices.Add("search", "Search");
+   actionChoices.Add("list", "List");
```

Comment:
The line starts on line 11 

```suggestion
```csharp {linenos=true,linenostart=11}
   // Top-level menu options
   Dictionary<string, string> actionChoices = new Dictionary<string, string>();
   actionChoices.Add("search", "Search");
   actionChoices.Add("list", "List");
```